### PR TITLE
Use DefaultAWSCredentialsProviderChain along with StaticCredentialsPr…

### PR DESCRIPTION
**Pull request by abhide**
…ovider

If secretKey or accessKey are blank in application.conf; then use DefaultAWSCredentialsProviderChain to figure out credentials and these credentials could be generated more dynamically.
https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.html

Tickets: https://github.com/vinyldns/vinyldns/issues/1008

Fixes #.1008

Changes in this pull request:
-
-

### CODE REVIEW

File: modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala [hunk 1]
```diff
@@ -71,23 +72,32 @@ class SqsMessageQueueProvider extends MessageQueueProvider {
...
           new EndpointConfiguration(
             sqsMessageQueueSettings.serviceEndpoint,
             sqsMessageQueueSettings.signingRegion
           )
         )
-        .withCredentials(
+      // If either of accessKey or secretKey are empty in conf file; then use AWSCredentialsProviderChain to figure out
+      // credentials.
+      // https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.html
+      if (StringUtils.isBlank(sqsMessageQueueSettings.accessKey) || StringUtils.isBlank(sqsMessageQueueSettings.secretKey)) {
```
Comment (pauljamescleary) [inline modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala:86]: what I would do is make the `SqsMessageQueueSesttings` for the access key and secret an `Option`, this way we can do...

File: modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala [hunk 2]
```diff
@@ -71,23 +72,32 @@ class SqsMessageQueueProvider extends MessageQueueProvider {
           s"signing region: ${sqsMessageQueueSettings.serviceEndpoint}; " +
           s"queue name: ${sqsMessageQueueSettings.queueName}"
       )
-      AmazonSQSAsyncClientBuilder
-        .standard()
-        .withEndpointConfiguration(
+
+      val sqsAsyncClientBuilder = AmazonSQSAsyncClientBuilder.standard()
+      sqsAsyncClientBuilder..withEndpointConfiguration(
```
Comment (pauljamescleary) [inline modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala:77]: looks like an extra dot here

File: modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala (hunk 1 above)
Comment (pauljamescleary) [reply modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala:86]: In `SqsMessageQueueSettings` ...

```
case class SqsMessageQueueSettings(
    accessKey: Option[String],
    secretKey: Option[String],
    serviceEndpoint: String,
    signingRegion: String,
    queueName: String
)
```

Then in here...
```
      val credProvider = sqsMessageQueueSettings.accessKey.zip(sqsMessageQueueSettings.secretKey).map {
        case (key, secret) =>
          new AWSStaticCredentialsProvider(
            new BasicAWSCredentials(key, secret)
          )
      }.headOption.getOrElse {
        new DefaultAWSCredentialsProviderChain()
      }

      sqsAsyncClientBuilder
        .withCredentials(credProvider)
        .build()
```

Comment (abhide) [reply modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala]: will take a look this suggestion.

File: modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala (hunk 2 above)
Comment (abhide) [reply modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala]: sorry.. my bad.. I didn't upload the latest changes. Should be fixed now

**Review by pauljamescleary — APPROVED** [review]

File: modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala (hunk 1 above)
Comment (abhide) [reply modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessageQueueProvider.scala]: Addressed here: https://github.com/vinyldns/vinyldns/pull/1014

**Pull request merged**